From f356dd60e79f972720a26a073c4fa0b100f0ca9a Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 12:01:57 -0400
Subject: [infiniband] mlx5: Use ARRAY_SIZE instead of sizeof/sizeof[0]

Message-id: <8246e95855e7ad39f0f9b2f1727ff90e2b4a0dc7.1410425016.git.avadai@redhat.com>
Patchwork-id: 93673
O-Subject: [Patch RHEL7.1 V1 181/197] IB/mlx5: Use ARRAY_SIZE instead of sizeof/sizeof[0]
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Fabian Frederick <fabf@skynet.be>

Acked-by: Eli Cohen <eli@mellanox.com>
Signed-off-by: Fabian Frederick <fabf@skynet.be>
Signed-off-by: Doug Ledford <dledford@redhat.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit a8f731ebd1b4f94cf52ff07fe524414b4fbf9793)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/mlx5/qp.c b/drivers/infiniband/hw/mlx5/qp.c
index 7efe6e3..8c574b6 100644
--- a/drivers/infiniband/hw/mlx5/qp.c
+++ b/drivers/infiniband/hw/mlx5/qp.c
@@ -2501,7 +2501,7 @@ int mlx5_ib_post_send(struct ib_qp *ibqp, struct ib_send_wr *wr,
  spin_lock_irqsave(&qp->sq.lock, flags);
 
  for (nreq = 0; wr; nreq++, wr = wr->next) {
-  if (unlikely(wr->opcode >= sizeof(mlx5_ib_opcode) / sizeof(mlx5_ib_opcode[0]))) {
+  if (unlikely(wr->opcode >= ARRAY_SIZE(mlx5_ib_opcode))) {
    mlx5_ib_warn(dev, "\n");
    err = -EINVAL;
    *bad_wr = wr;
-- 
1.7.1